python/ray/data/tests/test_map.py

```diff
@@ -330,6 +331,126 @@ def map_generator(item: dict) -> Iterator[int]:
     ]
 
 
+# Helper function to process timestamp data in nanosecs
+def process_timestamp_data(row):
+    # Convert numpy.datetime64 to pd.Timestamp if needed
+    if isinstance(row["timestamp"], np.datetime64):
+        row["timestamp"] = pd.Timestamp(row["timestamp"])
+
+    # Add 1ns to timestamp
+    row["timestamp"] = row["timestamp"] + pd.Timedelta(1, "ns")
+    return row
+
+
+# Helper function to create a sample timestamp nanosecs DataFrame
+def create_timestamp_dataframe():
+    return pd.DataFrame(
+        {
+            "id": [1, 2, 3],
+            "timestamp": pd.to_datetime(
+                [
+                    "2024-01-01 00:00:00.123456789",
+                    "2024-01-02 00:00:00.987654321",
+                    "2024-01-03 00:00:00.111222333",
+                ]
+            ),
+            "value": [10.123456789, 20.987654321, 30.111222333],
+        }
+    )
+
+
+# Handle Timestamp nanoseconds
+@pytest.mark.parametrize(
+    "df, expected_df",
+    [
+        (
+            create_timestamp_dataframe(),
```

Comment:
Unless I am missing something this looks identical to the result of `create_timestamp_dataframe`? Should we just call that function once? Should we only pass in one dataframe?

Reply:
Ah I see now that there is a 1ns difference, can we include a comment to specify that? Or could just have one DF passed in and do the addition of 1ns in the test itself?

Reply:
I will check if I can simplify this without limiting the correctness check.